src/menu.tsx

```diff
@@ -54,42 +62,50 @@ const Menu: React.FC<MenuProps> = ({renderTarget, renderMenu, width, position =
             return;
         }
 
-        const MARGIN_THRESHOLD = 12;
-        const {top: topTarget, width: widthTarget, height} = targetRect;
-        const top = topTarget + height;
-        const spaceTaken = parseInt(window.getComputedStyle(menu).getPropertyValue('height')) ?? 0;
-
-        const rightDirection = position === 'left' ? 'auto' : `calc(100% - ${widthTarget}px)`;
-
-        // if it doesn't fit on bottom
-        if (top + spaceTaken + MARGIN_THRESHOLD > window.innerHeight) {
-            const availableSpaceBottom = window.innerHeight - top;
-            if (topTarget /* this is the available space on top */ > availableSpaceBottom) {
-                setItemsComputedProps({
-                    right: rightDirection,
-                    bottom: '100%',
-                    top: 'auto',
-                    maxHeight: Math.min(topTarget, MAX_HEIGHT_DEFAULT),
-                    transformOrigin: 'center bottom',
-                });
-            } else {
-                setItemsComputedProps({
-                    top: '100%',
-                    bottom: 'auto',
-                    right: rightDirection,
-                    maxHeight: Math.min(window.innerHeight - top - MARGIN_THRESHOLD, MAX_HEIGHT_DEFAULT),
-                    transformOrigin: 'center top',
-                });
-            }
-        } else {
-            // if it fits on bottom
+        const {top: topTarget, width: widthTarget, left: leftTarget, bottom: bottomTarget} = targetRect;
+
+        const heightMenu = parseInt(window.getComputedStyle(menu).getPropertyValue('height')) ?? 0;
+
+        const leftDirection =
+            position === 'left' ? `${leftTarget}px` : `${leftTarget + widthTarget - width}px`;
+
+        const availableSpaceOnBottom = windowSize.height - bottomTarget - MARGIN_THRESHOLD;
+        const availableSpaceOnTop = topTarget - MARGIN_THRESHOLD;
+        const menuFitsOnBottom = availableSpaceOnBottom > heightMenu;
+        const menuFitsOnTop = availableSpaceOnTop > heightMenu;
+
+        if (menuFitsOnBottom) {
```

Comment:
previously we had 3 branches to draw the menu, now we have 4, is that new branch (if-else) needed?